mortar/src/main/java/mortar/bundler/BundleServiceRunner.java

```diff
@@ -105,4 +111,15 @@ void finishLoading() {
 
     state = State.IDLE;
   }
+
+  String bundleKey(MortarScope scope) {
+    if (rootScopePath == null) throw new IllegalStateException("Was this service not registered?");
+    String path = scope.getPath();
+    if (!path.startsWith(rootScopePath)) {
+      throw new IllegalArgumentException(String.format("\"%s\" is not under \"%s\"", scope,
+          rootScopePath));
+    }
+
+    return path.substring(rootScopePath.length());
```

Comment:
No, that's the bug. The path includes the activity scope name, which may include transient info like the task id. New process, new task id, all bundle info is lost.